dali/base/dadfs.cpp

```diff
@@ -2738,6 +2738,20 @@ class CDistributedFileBase : implements INTERFACE, public CInterface
         return *t;
     }
 
+    IPropertyTree &queryHistory()
+    {
+        IPropertyTree *t = root->queryPropTree("History");
+        if (!t)
+            t = root->setPropTree("History",createPTree("History")); // takes ownership
+        return *t;
+    }
+
+    void eraseHistory()
+    {
+        DistributedFilePropertyLock lock(this);
+        root->setPropTree("History",createPTree("History"));
```

Comment:
See other comments. Should handle no "History" node, so better to delete it here, not create blank.